hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -914,7 +914,7 @@ void checkMultiselect(List<? extends JpaSelection<?>> selections) {
 	@Override
 	public <T> SqmLiteral<T> literal(T value) {
 		if ( value == null ) {
-			return new SqmLiteralNull<>( this );
+			throw new IllegalArgumentException( "literal value cannot be null" );
```

Comment:
What do you think about protecting this with `JpaCompliance#isQueryComplianceEnabled`?